File: WitchcraftApp/src/test/java/com/tarbadev/witchcraft/recipes/domain/usecase/DoesRecipeExistUseCaseTest.java
package com.tarbadev.witchcraft.recipes.domain.usecase;

import com.tarbadev.witchcraft.recipes.domain.repository.RecipeRepository;
import com.tarbadev.witchcraft.recipes.domain.usecase.DoesRecipeExistUseCase;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.junit.MockitoJUnitRunner;

import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.BDDMockito.given;

@RunWith(MockitoJUnitRunner.class)
public class DoesRecipeExistUseCaseTest {
  @Mock
  private RecipeRepository recipeRepository;
  private DoesRecipeExistUseCase subject;

  @Before
  public void setUp() {
    subject = new DoesRecipeExistUseCase(recipeRepository);
  }

  @Test
  public void execute() {
    int id = 32;

    given(recipeRepository.existsById(id)).willReturn(true);

    assertThat(subject.execute(id)).isTrue();
  }
}

File: WitchcraftApp/src/test/kotlin/com/tarbadev/witchcraft/recipes/domain/usecase/DoesRecipeExistUseCaseTest.kt
package com.tarbadev.witchcraft.recipes.domain.usecase

import com.nhaarman.mockitokotlin2.mock
import com.nhaarman.mockitokotlin2.whenever
import com.tarbadev.witchcraft.recipes.domain.repository.RecipeRepository
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class DoesRecipeExistUseCaseTest {
    private val recipeRepository: RecipeRepository = mock()
    private lateinit var subject: DoesRecipeExistUseCase

    @BeforeEach
    fun setUp() {
        subject = DoesRecipeExistUseCase(recipeRepository)
    }

    @Test
    fun execute() {
        val id = 32

        whenever(recipeRepository.existsById(id)).thenReturn(true)

        assertTrue(subject.execute(id))
    }
}